### PR TITLE
Separate models

### PR DESCRIPTION
**Pull request by RannaAI**
Added a file called links.json which has all models
any model can be enabled/disabled and set either to be persistent or not through the start ui.

### CODE REVIEW

File: auto_ncpt.ipynb [hunk 1]
```diff
@@ -105,15 +106,165 @@
...
-    ")"
+    ")\n",
+    "\n",
+    "#default links\n",
+    "active_urls={\"models\":[],\"vae\":[],\"lora\":[],\"extensions\":[]}\n",
+    "\n",
+    "# links\n",
+    "links_file = 'links.json'\n",
+    "links={}\n",
+    "persistant_urls={\"models\":[],\"vae\":[],\"lora\":[],\"extensions\":[]}\n",
```
Comment (NoCrypt) [inline auto_ncpt.ipynb]: persistant_urls -> presistent_urls

File: links.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+    "models": [
+        "https://huggingface.co/NoCrypt/expermental_models/resolve/main/shux7.safetensors:enabled"
+    ],
+    "vae": [
+        "https://huggingface.co/NoCrypt/resources/resolve/main/VAE/any.vae.safetensors:enabled"
```
Comment (NoCrypt) [inline links.json]: i dunno if appending `:enabled` and `:keep` is a good idea in term of UX

**Review by NoCrypt — CHANGES_REQUESTED** [review]

File: auto_ncpt.ipynb [hunk 3]
```diff
@@ -105,15 +106,165 @@
     "    layout=layout,\n",
     ")\n",
     "\n",
+    "\n",
+    "output_control = Text(\n",
+    "    value=\"./outputs/\",\n",
+    "    description=\"Outputs path\",\n",
+    "    placeholder=\"hf_xyz\",\n",
```
Comment (NoCrypt) [inline auto_ncpt.ipynb]: change `placeholder` to be more meaningful, or remove it to avoid confusion

**Review by NoCrypt — CHANGES_REQUESTED** [review]

File: links.json (hunk 2 above)
Comment (RannaAI) [reply links.json]: Alright I will move them into separate arrays then, something like this :
`{
    "models": [
        "https://huggingface.co/NoCrypt/expermental_models/resolve/main/shux7.safetensors"
    ],
...
    "link_to_name": [],
    "enabled": ["https://huggingface.co/NoCrypt/expermental_models/resolve/main/shux7.safetensors"],
    "keep": []
}`

File: auto_ncpt.ipynb (hunk 3 above)
Comment (RannaAI) [reply auto_ncpt.ipynb]: fixed

File: links.json (hunk 2 above)
Comment (RannaAI) [reply links.json]: I made the change on the last commit 24e1e4d

File: auto_ncpt.ipynb [hunk 4]
```diff
@@ -38,56 +38,505 @@
...
+    "            button_style='info', \n",
+    "            tooltip='Zip folder and clean it up',\n",
+    "            icon='file' \n",
+    "        )\n",
+    "    \n",
+    "        #actions\n",
+    "        def clean_outputs(b):\n",
+    "            !rm -rf {b}\n",
+    "            b.mkdir(exist_ok=True)\n",
+    "            \n",
```
Comment (NoCrypt) [inline auto_ncpt.ipynb]: this method only used once and it's just 2 lines of code
maybe write it to [line 348](https://github.com/NoCrypt/paperspace-ncpt/pull/3/files#diff-5c0d7cf5ba0ad0bcb5fa6658592061099f7aeed8e1ccf9db62cb899d9465fcdbR348)

File: auto_ncpt.ipynb [hunk 5]
```diff
@@ -38,56 +38,505 @@
...
+    "            button_style='info', \n",
+    "            icon='recycle' \n",
+    "        )\n",
+    "    \n",
+    "        def clear_cached(b):\n",
+    "            display(\"clearing files...\")\n",
+    "            clear_button.style.button_color= \"orange\"\n",
+    "            !rm -rf '/notebooks/cached'\n",
+    "            clear_button.style.button_color= \"green\"\n",
+    "            clear_bin_button.description=\"Cleared!\"\n",
```
Comment (NoCrypt) [inline auto_ncpt.ipynb]: "clear_bin_button" is not defined or in another method

possible typo?

File: auto_ncpt.ipynb [hunk 6]
```diff
@@ -279,25 +756,23 @@
...
-    "if options.persistent_config :\n",
-    "    if not persistent_config_file.exists():\n",
+    "# deal with persistent config \n",
+    "if options.cached_config :\n",
+    "    if not cached_config_file.exists():\n",
     "        !cp {config_json} {notebook_dir}\n",
-    "    config_json=persistent_config_file\n",
-    "\n",
+    "    config_json=cached_config_file\n",
     "options.config_file= str(config_json)\n",
```
Comment (NoCrypt) [inline auto_ncpt.ipynb]: yeah this thing is read only :<

File: auto_ncpt.ipynb [hunk 7]
```diff
@@ -38,56 +38,505 @@
...
+    "import glob\n",
+    "\n",
     "\n",
+    "# Folder size method \n",
+    "def get_directory_size(directory_pattern):\n",
+    "    try:\n",
+    "        if '*' in directory_pattern:\n",
+    "            directories = glob.glob(directory_pattern)\n",
+    "        else:\n",
+    "            directories = [directory_pattern] if os.path.isdir(directory_pattern) else []\n",
```
Comment (NoCrypt) [inline auto_ncpt.ipynb]: if possible use `Path` instead of `os.path`

File: auto_ncpt.ipynb [hunk 8]
```diff
@@ -38,56 +38,505 @@
...
+    "\n",
+    "\n",
+    "    def __init__(self, control_list):\n",
+    "        self.controls = control_list\n",
+    "\n",
+    "    # load parameters from file otherwise it creates a default one \n",
+    "    def load(self, file = \"links.json\"):\n",
+    "        self.file_path = file\n",
+    "\n",
+    "        if not os.path.exists(self.file_path):\n",
```
Comment (NoCrypt) [inline auto_ncpt.ipynb]: if possible use `Path` instead of `os.path`

**Review by NoCrypt — CHANGES_REQUESTED** [review]

File: auto_ncpt.ipynb (hunk 4 above)
Comment (NoCrypt) [reply auto_ncpt.ipynb]: @RannaAI

**Review by NoCrypt — APPROVED** [review]

**Pull request merged**